sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallerInformationParametersShouldBeLast.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Method1(string callerFilePath) { }
+        public void Method2([CallerFilePath]string callerFilePath) { }
+        public void Method3(string other, [CallerFilePath]string callerFilePath) { }
+        public void Method4([CallerFilePath]string callerFilePath, string other) { } // Noncompliant {{Move 'callerFilePath' to the end of the parameter list.}}
+//                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        public void Method5([CallerFilePath]string callerFilePath, string other, [CallerLineNumber]int callerLineNumber) { }
+//                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        public void Method6(string first, [CallerFilePath]string callerFilePath, [CallerLineNumber]int callerLineNumber, string other) { }
+//                                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+//                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ @-1
+
+        public void Method7([CallerFilePath]string callerFilePath,
+//                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            [CallerLineNumber]int callerLineNumber,
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            [CallerMemberName]string callerMemberName, string other) { }
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+        public Program([CallerFilePath]string callerFilePath, string other) { } // Noncompliant
+        public Program(string other, [CallerFilePath]string callerFilePath) { }
+    }

```

Comment:
Please add a case where:
1. you override a method (we shouldn't report)
2. you implement an interface (we shouldn't report)

Reply:
@Evangelink that's valid cases, but I am not sure if we should not report here - you will still have the same problem, just you will not be able to fix it by reordering arguments... Perhaps the message should be different, wdyt?

Reply:
As you don't any control over how the base class/interface is defined I don't think we should report. If something is wrong on a tool you are using you shouldn't be impacted (your leak and your quality).

Reply:
done